extensions/adapters/vector/src/main/java/org/locationtech/geowave/adapter/vector/query/aggregation/VectorMathAggregation.java

```diff
@@ -53,38 +70,24 @@ public void setParameters(final FieldNameParam fieldNameParam) {
   }
 
   @Override
-  public BigDecimal merge(final BigDecimal result1, final BigDecimal result2) {
-    return agg(result1, result2);
+  public BigDecimal getResult() {
+    return value;
   }
 
   @Override
-  public BigDecimal getResult() {
-    return value;
+  public BigDecimal merge(final BigDecimal result1, final BigDecimal result2) {
+    return agg(result1, result2);
   }
 
+  // TODO: is it OK that I changed these? I fixed the tests.
   @Override
   public byte[] resultToBinary(BigDecimal result) {
-    if (result == null) {
-      return new byte[0];
-    }
-    final byte[] unscaled = result.unscaledValue().toByteArray();
```

Comment:
You can move this logic to the VarintUtils class since it is using Variable length encoding for each part of the BigDecimal.

Reply:
Talked about above, I moved this code over with some slight changes.